garrysmod/lua/includes/modules/markup.lua

```diff
@@ -350,52 +404,66 @@ function Parse( ml, maxwidth )
 							end
 						end
 
-						if ( lastSpacePos == string.len( curString ) && lastSpacePos > 0 ) then
-							local sequenceStartPos = utf8.offset( curString, 0, lastSpacePos )
-							ch = string.match( curString, utf8.charpattern, sequenceStartPos ) .. ch
-							j = utf8.offset( curString, 1, sequenceStartPos )
-							curString = string.sub( curString, 1, sequenceStartPos - 1 )
+						local previous_block = new_block_list[ #new_block_list ]
+						local wrap = lastSpacePos == string.len( curString ) && lastSpacePos > 0
+						if ( previous_block and previous_block.text:match(" $") and wrap and surface.GetTextSize( blk.text ) < maxwidth ) then
+							-- If the block was preceded by a space, wrap the block onto the next line first, as we can probably fit it there
+							local trimmed, trimCharNum = previous_block.text:gsub(" +$", "")
+							if ( trimCharNum > 0 ) then
+								previous_block.text = trimmed
+								previous_block.thisX = surface.GetTextSize( previous_block.text )
+							end
 						else
-							ch = string.sub( curString, lastSpacePos + 1 ) .. ch
-							j = lastSpacePos + 1
-							curString = string.sub( curString, 1, lastSpacePos )
-						end
+							if ( wrap ) then
+								-- If the block takes up multiple lines (and has no spaces), split it up
+								local sequenceStartPos = utf8.offset( curString, 0, lastSpacePos )
+								ch = string.match( curString, utf8.charpattern, sequenceStartPos ) .. ch
+								j = utf8.offset( curString, 1, sequenceStartPos )
+								curString = string.sub( curString, 1, sequenceStartPos - 1 )
+							else
+								-- Otherwise, strip the trailing space and start a new line
+								ch = string.sub( curString, lastSpacePos + 1 ) .. ch
+								j = lastSpacePos + 1
+								curString = string.sub( curString, 1, math.max( lastSpacePos - 1, 0 ) )
+							end
 
-						local m = 1
-						while string.sub( ch, m, m ) == " " do
-							m = m + 1
-						end
-						ch = string.sub( ch, m )
+							local m = 1
+							while string.sub( ch, m, m ) == " " do
+								m = m + 1
+							end
+							ch = string.sub( ch, m )
 
-						local x1,y1 = surface.GetTextSize( curString )
+							local x1,y1 = surface.GetTextSize( curString )
 
-						if ( y1 > thisMaxY ) then
-							thisMaxY = y1
-							ymaxes[ yOffset ] = thisMaxY
-							lineHeight = y1
-						end
+							if ( y1 > thisMaxY ) then
+								thisMaxY = y1
+								ymaxes[ yOffset ] = thisMaxY
+								lineHeight = y1
+							end
+
+							local new_block = {}

```

Comment:
Hello big sir,
Durga Software Solutions says you need to create tables this way

```lua
local new_block = {
	text = curString,
	font = blocks[i].font,
	colour = blocks[i].colour,
	thisY = thisY,
	thisX = x1,
	offset = {
		x = xOffset,
		y = yOffset
	}
}
```

thank u sir

Reply:
big yes